avogadro/io/trrformat.cpp

```diff
@@ -0,0 +1,446 @@
+                    mat[1][2] * NM_TO_ANGSTROM),
+            Vector3(mat[2][0] * NM_TO_ANGSTROM, mat[2][1] * NM_TO_ANGSTROM,
+                    mat[2][2] * NM_TO_ANGSTROM)));
+        }
+      }
+    }
+  }
+
+  typedef map<string, unsigned char> AtomTypeMap;
+  AtomTypeMap atomTypes;
+  unsigned char customElementCounter = CustomElementMin;
+
+  // Reading the coordinates of positions, velocities and forces
+  for (int _kid = 0; _kid < 3; ++_kid) {
+    natoms = header["natoms"];
+    double coords_double[DIM];
+    float coords_float[DIM];
+    for (int i = 0; i < natoms; ++i) {
+      if (header[key_check_2[_kid]] != 0) {
+        memset(coords_double, 0, sizeof(coords_double));
+        memset(coords_float, 0, sizeof(coords_float));
+        if (double_status) {
+          snprintf(fmt, sizeof(fmt), "%c%dd", endian, DIM);
+          inStream.read(buff, struct_calcsize(fmt));
+          struct_unpack(buff, fmt, &coords_double[0], &coords_double[1],
+                        &coords_double[2]);
+        } else {
+          snprintf(fmt, sizeof(fmt), "%c%df", endian, DIM);
+          inStream.read(buff, struct_calcsize(fmt));
+          struct_unpack(buff, fmt, &coords_float[0], &coords_float[1],
+                        &coords_float[2]);
+        }
+
+        if (key_check_2[_kid] == "x_size") {
+          // If parsed coordinates are fractional, the corresponding unscaling
+          // is done. Else the positions are assigned as parsed.
+          Vector3 pos(coords_double[0] * NM_TO_ANGSTROM +
+                        coords_float[0] * NM_TO_ANGSTROM,
+                      coords_double[1] * NM_TO_ANGSTROM +
+                        coords_float[1] * NM_TO_ANGSTROM,
+                      coords_double[2] * NM_TO_ANGSTROM +
+                        coords_float[2] * NM_TO_ANGSTROM);
+
+          AtomTypeMap::const_iterator it;
+          // if (it == atomTypes.end()) {
+          atomTypes.insert(
+            std::make_pair(to_string(i), customElementCounter++));
+          it = atomTypes.find(to_string(i));
+          // if (customElementCounter > CustomElementMax) {
+          //   appendError("Custom element type limit exceeded.");
+          //   return false;
+          // }
+          Atom newAtom = mol.addAtom(it->second);
+          newAtom.setPosition3d(pos);
+        }
+      }
+    }
+
+    // Set the custom element map if needed
+    if (!atomTypes.empty()) {
+      Molecule::CustomElementMap elementMap;
+      for (AtomTypeMap::const_iterator it = atomTypes.begin(),
+                                       itEnd = atomTypes.end();
+           it != itEnd; ++it) {
+        elementMap.insert(std::make_pair(it->second, "Atom " + it->first));
+      }
+      mol.setCustomElementMap(elementMap);
+    }
+  }
+  mol.setCoordinate3d(mol.atomPositions3d(), 0);
+
+  // Do we have an animation?
+  // EOF check
+  int coordSet = 1;
+  while (inStream.tellg() != file_len) {
```

Comment:
This seems to be where the compiler error on Windows is coming from - from memory I didn't think file lengths were expressed in int, but it looks like an ambiguous overload.